### PR TITLE
Use JSON and BSON in camelCase

### LINKED ISSUE #256: Add JSON metadata to each struct

### LINKED ISSUE #261: Change bson tags in each struct to use lower camel case

### PR DESCRIPTION
**Pull request by cirelli94**
Add JSON metadata to each struct, fix #256 
 Change BSON tags in each struct to use lower camel case, fix #261

### CODE REVIEW

File: model/model_test.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	//if err != nil {
+	//	t.Fail()
+	//}
+
+	//fmt.Println(string(out))
+
+	//var newCMS ClusterMembershipStatus
+	//newCMS.UnmarshalJSON(out)
+
+	//assert.Equal(t, cms, newCMS)
```
Comment (amreo) [inline model/model_test.go:50]: Why is this code commented?

File: model/oracle_database.go [hunk 2]
```diff
@@ -88,145 +88,145 @@ func (v *OracleDatabase) UnmarshalBSON(data []byte) error {
...
+		"platform",
+		"archivelog",
+		"charset",
+		"nCharset",
+		"blockSize",
+		"cpuCount",
+		"sgaTarget",
+		"pgatarget",
+		"memoryTarget",
+		"sgamaxSize",
```
Comment (amreo) [inline model/oracle_database.go:107]: m of max is in lowercase!

File: model/oracle_database.go [hunk 3]
```diff
@@ -88,145 +88,145 @@ func (v *OracleDatabase) UnmarshalBSON(data []byte) error {
...
+		"segmentsSize",
+		"datafileSize",
+		"allocated",
+		"elapsed",
+		"dbtime",
```
Comment (amreo) [inline model/oracle_database.go:112]: t o time is in lowercase!

File: model/oracle_database.go [hunk 4]
```diff
@@ -88,145 +88,145 @@ func (v *OracleDatabase) UnmarshalBSON(data []byte) error {
...
+		"isCDB",
+		"version",
+		"platform",
+		"archivelog",
+		"charset",
+		"nCharset",
+		"blockSize",
+		"cpuCount",
+		"sgaTarget",
+		"pgatarget",
```
Comment (amreo) [inline model/oracle_database.go:105]: t of target is in lowercase!

File: model/oracle_database.go [hunk 5]
```diff
@@ -88,145 +88,145 @@ func (v *OracleDatabase) UnmarshalBSON(data []byte) error {
...
+		"cpuCount": bson.M{
 			"bsonType": "string",
 			"minimum":  1,
 		},
-		"SGATarget": bson.M{
+		"sgaTarget": bson.M{
 			"bsonType": "number",
 		},
-		"PGATarget": bson.M{
+		"pgatarget": bson.M{
```
Comment (amreo) [inline model/oracle_database.go:193]: t of target is in lowercase!

File: model/oracle_database.go [hunk 6]
```diff
@@ -88,145 +88,145 @@ func (v *OracleDatabase) UnmarshalBSON(data []byte) error {
...
 			"bsonType": "number",
 		},
-		"MemoryTarget": bson.M{
+		"memoryTarget": bson.M{
 			"bsonType": "number",
 		},
-		"SGAMaxSize": bson.M{
+		"sgamaxSize": bson.M{
```
Comment (amreo) [inline model/oracle_database.go:199]: m of max is in lowercase!

File: model/oracle_database.go [hunk 7]
```diff
@@ -88,145 +88,145 @@ func (v *OracleDatabase) UnmarshalBSON(data []byte) error {
...
 		},
-		"Elapsed": bson.M{
+		"elapsed": bson.M{
 			"anyOf": bson.A{
 				bson.M{"type": "null"},
 				bson.M{"type": "number"},
 			},
 		},
-		"DBTime": bson.M{
+		"dbtime": bson.M{
```
Comment (amreo) [inline model/oracle_database.go:217]: t of time is in lowercase!

File: model/oracle_exadata_component.go [hunk 8]
```diff
@@ -73,71 +73,71 @@ func (v *OracleExadataComponent) UnmarshalBSON(data []byte) error {
...
-		"ServerType": bson.M{
+		"serverType": bson.M{
 			"bsonType": "string",
 			"enum": bson.A{
-				"StorageServer",
-				"DBServer",
-				"IBSwitch",
+				"storageServer",
+				"dbServer",
+				"ibSwitch",
```
Comment (amreo) [inline model/oracle_exadata_component.go:111]: Can you revert these changes?

File: model/oracle_exadata_component.go [hunk 9]
```diff
@@ -149,56 +149,56 @@ var OracleExadataComponentBsonValidatorRules = bson.M{
...
+		"flashcacheMode": bson.M{
 			"anyOf": bson.A{
 				bson.M{"bsonType": "null"},
 				bson.M{
 					"bsonType": "string",
 					"enum": bson.A{
-						"WriteBack",
-						"WriteThrough",
+						"writeBack",
+						"writeThrough",
```
Comment (amreo) [inline model/oracle_exadata_component.go:196]: Can you revert these changes?

File: model/technology_status.go [hunk 10]
```diff
@@ -17,10 +17,10 @@ package model
...
 type TechnologyStatus struct {
 	Product    string
-	Used       float64 `json:"-"`
-	Count      float64 `json:"-"`
-	TotalCost  float64 `json:"-"`
-	PaidCost   float64 `json:"-"`
+	Used       float64 `json:"-" bson:"-"`
+	Count      float64 `json:"-" bson:"-"`
+	TotalCost  float64 `json:"-" bson:"-"`
+	PaidCost   float64 `json:"-" bson:"-"`
```
Comment (amreo) [inline model/technology_status.go:23]: TechnologyStatus don't need to be marshalled to BSON

File: model/hostdata_fe_schema_validator.go [hunk 11]
```diff
@@ -15,43 +15,45 @@
...
-        "SchemaVersion",
-        "Info",
-        "ClusterMembershipStatus",
-        "Features",
-        "Filesystems"
+        "hostname",
+        "location",
+        "environment",
+        "tags",
+        "agentVersion",
```
Comment (amreo) [inline model/hostdata_fe_schema_validator.go:30]: Can you update the string?

File: model/hostdata_be.go [hunk 12]
```diff
@@ -29,23 +29,23 @@ const ServerSchemaVersion int = 1
...
+	Location                string                  `json:"location" bson:"location"`
+	Environment             string                  `json:"environment" bson:"environment"`
+	AgentVersion            string                  `json:"agentVersion" bson:"agentVersion"`
+	Tags                    []string                `json:"tags" bson:"tags"`
+	Info                    Host                    `json:"info" bson:"info"`
+	ClusterMembershipStatus ClusterMembershipStatus `json:"clusterMembershipStatus" bson:"clusterMembershipStatus"`
+	Features                Features                `json:"features" bson:"features"`
+	Filesystems             []Filesystem            `json:"filesystems" bson:"filesystems"`
+	Clusters                []ClusterInfo           `json:"clusters" bson:"clusters"`
+	OtherInfo               map[string]interface{}  `json:"otherInfo" bson:"otherInfo"`
```
[the remark this answers is not in the document]
Comment (cirelli94) [reply model/hostdata_be.go:48]: Originally, it was `bson:"OtherInfo"`

**Comment by amreo** [comment]
model/hostdata.go wasn't updated!

File: model/oracle_database.go (hunk 4 above)
Comment (cirelli94) [reply model/oracle_database.go:105]: :facepalm:

**Comment by cirelli94** [comment]
@amreo what should be updated in `model/hostdata.go`?

**Comment by amreo** [comment]
The fields name in json (json:"...")

**Comment by amreo** [comment]
Fix #260

**Comment by cirelli94** [comment]
Fix #260

**Comment by amreo** [comment]
Fix #253
